lib/util/precision.js

```diff
@@ -0,0 +1,47 @@
+const Big = require('bignumber.js')
+
+const DEFAULT_SIG_FIGS = 5
+const PRICE_SIG_FIGS = 5
+const AMOUNT_DECIMALS = 8
+
+/**
+ * Smartly set the precision (decimal) on a value based off of the significant
+ * digit maximum. For example, calling with 3.34 when the max sig figs allowed
+ * is 5 would return '3.3400', the representation number of decimals IF they
+ * weren't zeros.
+ *
+ * @param {number} n
+ * @param {number} maxSigs - default 5
+ * @return {string} str
+ */
+const setSigFig = (number = 0, maxSigs = DEFAULT_SIG_FIGS) => {
+  const n = +(number)
+  if (!isFinite(n)) {
+    return number
+  }
+  const value = n.toPrecision(maxSigs)
+
+  return value.match(/e/)
```

Comment:
You can use `/e/.test(value)` here, which returns a boolean and is faster.

In which case `e` will be present in the number? a test would be nice